src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,44 +1,89 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todos: Todo[],
+  setModal: (val: boolean) => void,
+  id: number,
+  setId: (val: number) => void,
+  loader: boolean,
+  setLoader: (val: boolean) => void,
+  userId: number,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todos,
+  setModal,
+  id,
+  setId,
+  loader,
+  setLoader,
+  userId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setLoader(true);
+
+    getUser(userId)
+      .then((userData) => {
+        setUser(userData);
+      })
+      .catch(() => {})
+      .finally(() => setLoader(false));
+  }, [setLoader, userId]);
+
+  const handleCloseModal = useCallback(() => {
+    setModal(false);
+    setId(0);
+  }, [setModal, setId]);
+
+  const getTodoById = useCallback((todoId: number) => {
```

Comment:
They're a lot of unnecessary useCallback in the code. If the function doesn't pass to another component like a prop it will be not necessary to write it. Also it decreases readability